### ISSUE TITLE
Use sklearn.model_selection.ShuffleSplit to create subsamples?

### ISSUE THREAD

**Issue by Vayel** [MaxHalford/ethik#69]
https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.ShuffleSplit.html

Instead of calling `yield_masks(n_masks, n, p)`, we would call:

```python
rs = ShuffleSplit(n_splits=n_masks, train_size=p)
for sample_index, _ in rs.split(X_test):
    pass
```

@MaxHalford ?
